Src/Microsoft.Dynamic/Runtime/BinderOps.cs

```diff
@@ -152,6 +152,10 @@ public static class BinderOps {
             return new ArgumentTypeException($"{name}() got multiple values for keyword argument '{argumentName}'");
         }
 
+        public static ArgumentTypeException TypeErrorForDuplicateArgument(string name, int position, string argumentName) {
+            return new ArgumentTypeException($"Argument for {name}() given by name ('{argumentName}') and position ({position})");
```

Comment:
I was aware of that. I didn't elaborate that in my initial description because it was already long and convoluted...

I chose the capital "A" because it is the DLR and all (most?) messages it produces are properly sentence-capitalized. This is also the convention throughout .NET. I don't particularly care either, as long as the messages are clear, unambiguous, fairly precise, helpful, and accurate. Also, the same error condition can be perceived and reported from various angles. For instance, what if there is a positional, and two keyword arguments clashing? I think it doesn't matter as long as the message is correct from some point.

CPython is also not consistent in its choices of reporting perspectives. As a teaser, the example I gave in my PR description looks differently in 3.9:

~~~
Python 3.9.0 (tags/v3.9.0:9cf6752, Oct  5 2020, 15:34:40) [MSC v.1927 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import datetime
>>> datetime.date(year=2020, *(11, 2))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: function missing required argument 'day' (pos 3)
~~~